sdk/search/Azure.Search.Documents/src/Models/SearchMode.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// Specifies whether any or all of the search terms must be matched in order to count the document as a match.
+    /// </summary>
+    [CodeGenSchema("SearchMode")]
+    public enum SearchMode
+    {
+        /// <summary>
+        /// Any search terms may match.
+        /// </summary>
+        Any,
```

Comment:
Should we explicitly mark this as ` = 0`?

Reply:
Why? It's already 0 and always will be. The serializer doesn't. I only had to do this to fix an issue where `SearchMode` exists in both searchindex.json and searchservice.json but they are different, and the generator treats the latter one as a constant anyway - or does correctly, once I fix this.

Reply:
I like it as an explicit hint, but I guess ApiCompat would check if folks (or switching to the generator) would someday reorder this.